.github/workflows/hive-ci.yml

```diff
@@ -19,22 +19,12 @@
 
 name: "Hive CI"
 on:
-  push:
-    paths-ignore:
-    - '.github/workflows/python-ci.yml'
-    - '.github/workflows/spark-ci.yml'
-    - '.github/workflows/flink-ci.yml'
-    - '.github/workflows/cancel-duplicate-workflow-runs.yml'
-    - '.gitignore'
-    - 'dev/**'
-    - 'arrow/**'
-    - 'spark/**'
-    - 'flink/**'
-    - 'pig/**'
-    - 'python/**'
-    - 'python_legacy/**'
-    - 'site/**'
   pull_request:
+    types:
+      - opened
+      - reopened
+      - ready_for_review
+      - synchronize
```

Comment:
One of the benefits of `synchronize` is that we don't have to duplicate the list of paths to ignore in each workflow file.

Currently, we have them both in `on: push` and `on: pull_request`. I might open a separate PR that adds this in earlier just to simplify things. Apparently `on: push` is discouraged for projects that are commonly used as templates for forks anyway. But the main driver is just removing the large block of ignore paths.

Reply:
For testing the build _after_ its been merged into master, we want those to always go to completion (which they don't).

Currently, this is handled by a mixture of `on: push` and the cancel duplicates workflow.

I believe I can fix it so that tests run, to completion, for every single PR that gets merged (so they show green or red correctly).